src/main/java/io/confluent/kafkarest/MetadataObserver.java

```diff
@@ -63,6 +63,22 @@ public MetadataObserver(KafkaRestConfig config, ZkClient zkClient) {
     return brokerIds;
   }
 
+  public Broker getBrokerById(final int brokerId) {
```

Comment:
A couple of notes on this metadata code:

1. We may need to start doing some caching. This is tricky since then we need some sort of invalidation interface as well, i.e. if the SimpleConsumer tries to use the leader returned and finds out it is no longer the leader it should invalidate the cache and retry the whole process before failing.
2. I know the other code use ZkUtils, but we want to start minimizing our dependence on the server internals where possible. We can get all the info we need here from a MetadataRequest. Might be better to reuse some of the code in the kafka-clients library (i.e. the Metadata, Cluster, and some associated classes), although the metadata stuff isn't nicely separated out yet.
3. One drawback to adding this now is that we don't have access to the new consumer, which would be able to get all this information for you and we wouldn't even need this code (at least for this; we may need it for APIs that show the state of the cluster). And we're likely going to convert over to the new consumer anyway since it should be released in the next Kafka release and available by the time we tag our next release.

Not sure yet how to best handle this transition and code reuse. It might end up making sense to use the code as is or with only some caching added and file the rest of it as follow up issues.